### PR TITLE
[FIX][CORE] #235 Fix activity dispatching for partial sessions

### PR DESCRIPTION
**Pull request by tobous**
Now correctly updates the ProjectMapper for partially shared resources
when dispatching activities.

This patch was provided by @srossbach.

### CODE REVIEW

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java [hunk 1]
```diff
@@ -754,9 +754,20 @@ private void sendActivity(final List<User> recipients,
...
+        /*
+         * To explain some magic: When the host generates an activity this
+         * method will be called multiple times. It is first called with the
+         * host as it is only recipient and then it is called again for the
+         * other recipients.
+         */
+        // handle IFileSystemModificationActivities to update ProjectMapper
         if (activity instanceof IFileSystemModificationActivity) {
-            send = updatePartialSharedResources((IFileSystemModificationActivity) activity);
+            if (!isHost() || (isHost() && recipients
```
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java]: Code Smell Code Smell: Merge this if statement with the enclosing one. ([squid:S1066](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1066))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=237&issues=AWYXZoXdM63is4RDRNN2&open=AWYXZoXdM63is4RDRNN2)

[ISSUE_KEY]: AWYXZoXdM63is4RDRNN2

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java [hunk 2]
```diff
@@ -754,9 +754,20 @@ private void sendActivity(final List<User> recipients,
         }
 
         boolean send = true;
-        // handle FileActivities and FolderActivities to update ProjectMapper
+
+        /*
+         * To explain some magic: When the host generates an activity this
```
Comment (srossbach) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java]: We should also move this check and the one above to line 159.

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java (hunk 1 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java]: Code Smell Code Smell: Merge this if statement with the enclosing one. ([squid:S1066](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1066))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=237&issues=AWYXZoXdM63is4RDRNN2&open=AWYXZoXdM63is4RDRNN2)

[ISSUE_KEY]: AWYXZoXdM63is4RDRNN2

**Comment by tobous** [comment]
@srossbach can I merge this pull request or does it require changes regarding the other issue pointed out in #235?

**Comment by srossbach** [comment]
Yes you can merge this PR. The other issue is a complete different topic.

File: de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java (hunk 1 above)
Comment (saros-infrastructure) [inline de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SarosSession.java]: Code Smell Code Smell: Merge this if statement with the enclosing one. ([squid:S1066](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1066))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=237&issues=AWYXZoXdM63is4RDRNN2&open=AWYXZoXdM63is4RDRNN2)

[ISSUE_KEY]: AWYXZoXdM63is4RDRNN2

**Comment by tobous** [comment]
@srossbach Could you approve this again? I just merged the two if clauses.

**Pull request merged**